### ISSUE TITLE
Sidebar collapse state memory inconsistant

### ISSUE THREAD

**Issue by bananafeller** [marktext/marktext#2421]
### Description

Different behaviour when using Ctrl+J or the buttons ( ![mark text folder](https://user-images.githubusercontent.com/54718562/100907033-b27d1e80-347e-11eb-8cf2-2f89440d5c7b.png), ![mark text Search](https://user-images.githubusercontent.com/54718562/100907029-b1e48800-347e-11eb-933a-4ab9631073c9.png), ![mark text ToC](https://user-images.githubusercontent.com/54718562/100907031-b27d1e80-347e-11eb-97c3-a62280132894.png) ) to show/hide sidebar.

This bug has been mentioned in #2268 and #2345, however neither used the bug template and they both describe different but related issues. 

- [x] Can you reproduce the issue? 

#### Using Buttons ( ![mark text folder](https://user-images.githubusercontent.com/54718562/100907033-b27d1e80-347e-11eb-8cf2-2f89440d5c7b.png), ![mark text Search](https://user-images.githubusercontent.com/54718562/100907029-b1e48800-347e-11eb-933a-4ab9631073c9.png), ![mark text ToC](https://user-images.githubusercontent.com/54718562/100907031-b27d1e80-347e-11eb-97c3-a62280132894.png) ) 
1. Do any of the following

   - Change the width of the sidebar
   - Collapse "Opened files" or "Entire Open Directory" in the Folder section ( ![mark text folder](https://user-images.githubusercontent.com/54718562/100907033-b27d1e80-347e-11eb-8cf2-2f89440d5c7b.png) )
   - Collapse any of the headings in the Table of Contents section ( ![mark text ToC](https://user-images.githubusercontent.com/54718562/100907031-b27d1e80-347e-11eb-97c3-a62280132894.png) )

2. Press which ever button is active → Sidebar will disappear
3. Press any one of the buttons → Sidebar will reappear 

**Expected behavior:**
What ever change was made in step 1 should remain.

**Actual behavior:**
The changes reset to default

#### Using "Ctrl+J"
1. Do any of the following

   - Change the width of the sidebar
   - Collapse "Opened files" or "Entire Open Directory" in the Folder section ( ![mark text folder](https://user-images.githubusercontent.com/54718562/100907033-b27d1e80-347e-11eb-8cf2-2f89440d5c7b.png) )
   - Collapse any of the headings in the Table of Contents section ( ![mark text ToC](https://user-images.githubusercontent.com/54718562/100907031-b27d1e80-347e-11eb-97c3-a62280132894.png) )

2. Press Ctrl+J → Sidebar will disappear
3. Press Ctrl+J → Sidebar will reappear 

**Actual behavior:**
This provides the desired behaviour, the changes are retained. 

#### Notes
- Changes to the collapsed/expanded folders within the Opened Directory are retained using either Ctrl+J or the buttons.
- The sidebar width is retained if Mark Text is restarted.

**Link to an example:**

![Sidebar bug](https://user-images.githubusercontent.com/54718562/100913797-1a376780-3487-11eb-9319-fcba4d479de3.gif)

### Versions

- Mark Text version: v0.16.2
- Operating system: Windows 10 Pro, Version 1909, Build 18363.1198

**Comment by MintYale** [comment]
Yes, the status of sidebar should remain, and it should better be cached and saved so that even the app is closed and then reopened next time, the status bar is still the same as before. Many apps such as Typora/VS Code act like this. It's much easier to use. MarkText is almost perfect for me except this issue. Hope it can be fixed someday.

**Comment by rkristelijn** [comment]
Please see https://github.com/marktext/marktext/pull/4345

<img width="892" height="578" alt="Image" src="https://github.com/user-attachments/assets/b82ad505-8c3b-4c14-aec1-aeaf750c9b07" />
